### PR TITLE
Merge additional pydocstyle error fixes into development docs

### PR DESCRIPTION
**Pull request by RNKuhns**
#### Reference Issues/PRs

This partially addresses docstring updates needed to eventually implement #890. 

#### What does this implement/fix? Explain your changes.

Primarily includes docstring fixes to pass `pydocstyle` checks that were needed because changes had to be made to some docstrings for `numpydoc` parsing.

Also includes a couple tweaks to the look of the documentation pages. 

- Changes navigation bar titles from users to "user documentation" and developers to "get involved"
- Makes updates to "getting started" page
- Turns on autodoc member ordering by source to order documentation of members (eg. attributes, methods) by the order they show up in source code (not sure this is working)

#### Does your contribution introduce a new dependency? If yes, which one?

N/A. Just updates docstrings. 

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by RNKuhns** [comment]
@mloning can you take a look at the changes in this and merge it into our development documentation branch if they look fine?

Almost exclusively updates to pass `pydocstyle` checks. Also included minor tweaks to performance metric documentation's .rst file.

**Comment by aiwalter** [comment]
great work @RNKuhns

**Comment by mloning** [comment]
Fixed the merge conflicts

File: docs/source/conf.py [hunk 1]
```diff
@@ -97,7 +97,17 @@
...
-autodoc_default_options = {"members": True, "inherited-members": True}
+
+# members and inherited-members default to showing methods/attributes, etc
+# from a class that are created in the class or inherited
+# member-order says to order the documentation in the order the members
+# were defined in the source code
+autodoc_default_options = {
+    "members": True,
+    "inherited-members": True,
+    "member-order": "bysource",
```
Comment (mloning) [inline docs/source/conf.py:108]: this is nice! so we should make sure `fit`, `predict`, `transform` and `update` are at the top in the code, right?

**Review by mloning — APPROVED** [review]
Great work @RNKuhns - looks all good to me

**Review by fkiraly — APPROVED** [review]
Nice and also fixes docstrings in different places

**Comment by mloning** [comment]
I've merged this now @RNKuhns - wanted to look at a few other things based on your changes!

**Pull request merged**